gems/kaminari-core/1.2/kaminari-core.rbs

```diff
@@ -29,6 +29,16 @@ module Kaminari
     def out_of_range?: () -> bool
   end
 
+  module ConfigurationMethods
+    module ClassMethods
+      def paginates_per: (Integer val) -> void
+      def default_per_page: () -> Integer
+      def max_paginates_per: (Integer val) -> void
+      def max_per_page: () -> Integer?
+      def max_pages: (?Integer | Symbol val) -> (Integer? | void)
```

Comment:
How about the following definition?

```suggestion
      def max_pages: (Integer) -> void
                   | () -> Integer
```

The documentation says:

* Overrides the max_pages value per model *when a value is given*
* Also returns this model's max_pages value (snip) *when no value is given*

It means this method has different behaviors depending on whether it has arguments or not. So I think the type definition should represent them.

The `:none` is a marker for no arguments. It does not mean this method receives a Symbol.

https://github.com/kaminari/kaminari/blob/bf20fccfdbf7d10c337aeead7ecf4f54a230a5fb/kaminari-core/lib/kaminari/models/configuration_methods.rb#L37-L52